docs/users.md

```diff
@@ -314,6 +316,12 @@ To create a control plane of type `host` run the command:
 kflex create cp4 --type host
 ```
 
+To create a control plane of type `external` with the required options, run the command: 
+
+```shell
+kflex adopt --adopted-context <kubeconfig-context-of-external-cluster> cp5
```

Comment:
Could a separated kubeconfig be passed with `KUBECONFIG=... kflex...` or `kflex --kubeconfig <file>'?

Reply:
As this command both create the secret and create the CP... it would be hard to use in combination with KS future Helm chart. The process would be
1. create/push the secret in the cluster manually or via kflex.
2. tell KS helm to create a CP using it (either upon installation or later on)

Reply:
> Could a separated kubeconfig be passed with `KUBECONFIG=... kflex...` or `kflex --kubeconfig '?

Yes. These are all the options for the adopt subcommand. You'd be looking at the `--adopted-kubeconfig` flag. 

```
$ kflex adopt --help
Adopt a control plane from an external cluster and switches the Kubeconfig context to
                the current instance

Usage:
  kflex adopt <name> [flags]

Flags:
  -c, --adopted-context string           path to adopted cluster context in adopted kubeconfig
  -a, --adopted-kubeconfig string        path to adopted cluster kubeconfig file. If unspecified, it uses the default Kubeconfig
  -s, --chatty-status                    chatty status indicator (default true)
  -x, --expiration-seconds int           adopted token expiration in seconds. Default is one year. (default 31536000)
  -h, --help                             help for adopt
  -k, --kubeconfig string                path to kubeconfig file
  -p, --postcreate-hook string           name of post create hook to run
  -e, --set stringArray                  set post create hook variables, in the form name=value 
  -d, --skip-url-override                skip URL override, used for local debugging only
  -u, --url-override https://127.0.0.1   URL overrride for adopted cluster. Required when cluster address uses local host address, e.g. https://127.0.0.1 
  -v, --verbosity int                    log level
  ```

Reply:
> if no value is passed to --adopted-context could it default to the current one... especially if one passes a separate kubeconfig with only one context in it?

Currently you do need to pass the context. It might be confusing in the common developer scenario where there is one kubeconfig with a context for each cluster, since the "current" context is typically the kubeflex hosting cluster context. It might be possible when a different Kubeconfig is used (which needs to be specified with `--adopted-kubeconfig`) to check if there is only one context and in that case not require the use of `--adopted-context`. But I would leave this as a future feature/PR if we really think it is useful.

Reply:
> As this command both create the secret and create the CP... it would be hard to use in combination with KS future Helm chart...

One possible approach for using kflex to create bootstrap secret and the helm chart to create the CP would be to add an option to the `adopt` subcommand `--create-bootstrap-secret-only` - would that help?

Reply:
> > As this command both create the secret and create the CP... it would be hard to use in combination with KS future Helm chart...
> 
> One possible approach for using kflex to create bootstrap secret and the helm chart to create the CP would be to add an option to the `adopt` subcommand `--create-bootstrap-secret-only` - would that help?

Assuming that one has a kubeconfig for the external cluster, I guess one can use kubectl to do it... that's way I was asking for clarifications on what the secret contains

```shell
kubectl create secret generic <cluster1-bootstrap-secret> --from-file=<cluster1-key>=<cluster1-kubeconfig-path> --namespace <cluster1-bootstrap-namespace>
```

Reply:
Since it is assumed that the bootstrap kubeconfig has only one context, you probably need first to extract a new kubeconfig only for the context of the adopted cluster, so you'd need to run first a command such as:

```bash
kubectl --kubeconfig=$SOURCE_KUBECONFIG config view --minify --flatten \
--context=$DESIRED_CONTEXT > $OUTPUT_KUBECONFIG
```

and then you can run the `kubectl create secret generic ...`  command

Reply:
Since a kubeconfig has a "current context", the natural default for `--adopted-context` is the current context of the adopted kubeconfig.